### PR TITLE
added mention in edit post and fix some issues

### PR DESCRIPTION
**Pull request by NaingAmity**

### CODE REVIEW

File: src/components/EditPostModal/index.tsx [hunk 1]
```diff
@@ -111,6 +116,7 @@ const EditPostModal = ({
   };
 
   const handleEditPost = async () => {
+    const mentionees = mentionUsers.map((user) => user.id);
     if (displayImages.length > 0) {
       const fileIdArr: (string | undefined)[] = displayImages.map(
```
Comment (frankent) [inline src/components/EditPostModal/index.tsx:121]: Should be this?

```suggestion
      const fileIdArr: string[] | undefined = displayImages.map(
```

File: src/components/EditPostModal/index.tsx [hunk 2]
```diff
@@ -257,13 +267,22 @@ const EditPostModal = ({
...
+            <ScrollView
+              style={styles.container}
+              keyboardShouldPersistTaps="handled"
+            >
+              <MentionInput
                 style={styles.textInput}
-                value={inputMessage}
-                onChangeText={(text) => setInputMessage(text)}
+                isBottomMentionSuggestionsRender={true}
+                placeholder="What's going on...?"
```
Comment (frankent) [inline src/components/EditPostModal/index.tsx:277]: Do we have i18n ?

File: src/components/UserItem/index.tsx [hunk 3]
```diff
@@ -11,12 +11,14 @@ export default function UserItem({
   showThreeDot,
   onPress,
   onThreeDotTap,
+  hideMenu,
 }: {
   user: UserInterface;
   isCheckmark?: boolean | undefined;
   showThreeDot?: boolean | undefined;
```
Comment (frankent) [inline src/components/UserItem/index.tsx:18]:
```suggestion
  isCheckmark?: boolean;
  showThreeDot?: boolean;
```

File: src/providers/Social/feed-sdk.ts [hunk 4]
```diff
@@ -162,15 +162,28 @@ export async function createPostToFeed(
 export async function editPost(
   postId: string,
   content: { text: string; fileIds: string[] },
-  postType: string
+  postType: string,
+  mentionees: string[],
+  mentionPosition: IMentionPosition[]
 ): Promise<Amity.Post<any>> {
-  let postParam = {};
+  console.log('GG', mentionees);
```
Comment (frankent) [inline src/providers/Social/feed-sdk.ts:169]:
```suggestion
```

File: src/providers/Social/feed-sdk.ts [hunk 5]
```diff
@@ -162,15 +162,28 @@ export async function createPostToFeed(
...
+  let postParam = {
+    mentionees:
+      mentionees.length > 0
+        ? ([
+            { type: 'user', userIds: mentionees },
+          ] as Amity.MentionType['user'][])
+        : [],
+    metadata: { mentioned: mentionPosition },
+  };
+  console.log(postParam);
```
Comment (frankent) [inline src/providers/Social/feed-sdk.ts:179]:
```suggestion
```

File: src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx [hunk 6]
```diff
@@ -79,8 +79,10 @@ const CommunityMembersTab: React.FC<ICommunityMembersTab> = ({
     }, [getCommunityMemberList, shouldRefresh])
   );
 
-  const renderMember = ({ item }: { item: Amity.Member<'community'> }) => {
+  const renderMember = ({ item }: { item: Amity.Membership<'community'> }) => {
+    console.log(item.roles);
```
Comment (frankent) [inline src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx:83]:
```suggestion
```

**Review by frankent — CHANGES_REQUESTED** [review]
Change request krub, please check all my comments and questions

File: src/components/EditPostModal/index.tsx (hunk 1 above)
Comment (NaingAmity) [reply src/components/EditPostModal/index.tsx:121]: I checked current code structure and we need undefined array. 
These undefined will be handled in editPost function.

File: src/components/EditPostModal/index.tsx (hunk 2 above)
Comment (NaingAmity) [reply src/components/EditPostModal/index.tsx:277]: We had already installed that package. But not setup properly yet.

File: src/components/UserItem/index.tsx (hunk 3 above)
Comment (NaingAmity) [reply src/components/UserItem/index.tsx:18]: done krub

File: src/providers/Social/feed-sdk.ts (hunk 5 above)
Comment (NaingAmity) [reply src/providers/Social/feed-sdk.ts:179]: done

File: src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx (hunk 6 above)
Comment (NaingAmity) [reply src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx:83]: done

File: src/components/EditPostModal/index.tsx (hunk 1 above)
Comment (frankent) [reply src/components/EditPostModal/index.tsx:121]: ok

**Review by frankent — APPROVED** [review]
LGTM

**Pull request merged**